core/src/main/java/com/graphhopper/routing/util/parsers/OSMMaxSpeedParser.java

```diff
@@ -39,27 +39,19 @@ public OSMMaxSpeedParser(DecimalEncodedValue carMaxSpeedEnc) {
 
     @Override
     public void handleWayTags(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWay way, IntsRef relationFlags) {
-        double maxSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed"));
-        double fwdSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed:forward"));
-        if (!isValidSpeed(fwdSpeed) && isValidSpeed(maxSpeed))
-            fwdSpeed = maxSpeed;
-        double maxPossibleSpeed = MaxSpeed.UNLIMITED_SIGN_SPEED;
-        if (isValidSpeed(fwdSpeed) && fwdSpeed > maxPossibleSpeed)
-            fwdSpeed = maxPossibleSpeed;
-
-        double bwdSpeed = OSMValueExtractor.stringToKmh(way.getTag("maxspeed:backward"));
-        if (!isValidSpeed(bwdSpeed) && isValidSpeed(maxSpeed))
-            bwdSpeed = maxSpeed;
-        if (isValidSpeed(bwdSpeed) && bwdSpeed > maxPossibleSpeed)
-            bwdSpeed = maxPossibleSpeed;
-
-        if (!isValidSpeed(fwdSpeed))
-            fwdSpeed = UNSET_SPEED;
-        carMaxSpeedEnc.setDecimal(false, edgeId, edgeIntAccess, fwdSpeed);
+        carMaxSpeedEnc.setDecimal(false, edgeId, edgeIntAccess, getMaxSpeed(way, false));
+        carMaxSpeedEnc.setDecimal(true, edgeId, edgeIntAccess, getMaxSpeed(way, true));
+    }
```

Comment:
In contrast to the AverageSpeedParser the OSMMaxSpeedParser does not neglect `maxspeed:foward/backward` when it is larger than `maxspeed`. I also tried to make this method more readable, it should still do the same.